src/rebar_prv_dialyzer.erl

```diff
@@ -382,6 +384,29 @@ run_dialyzer(State, Opts) ->
             {0, State}
     end.
 
+filter_warnings([], Warnings) ->
+    Warnings;
+filter_warnings(Ignores, Warnings) ->
+    lists:filter(fun(Warning) ->
+        not lists:any(fun (Ignore) -> match_warning(Ignore, Warning) end, Ignores)
+    end, Warnings).
+
+match_warning({MatchTag, {MatchPath, MatchLine}, MatchMsg}, {Tag, {Path, Line}, Msg}) when MatchPath =/= '_' ->
+    lists:suffix(MatchPath, Path) andalso match_term([MatchTag, MatchLine, MatchMsg], [Tag, Line, Msg]);
```

Comment:
@tsloughter how'd you feel about going for relative paths from the root in rebar3?

I'm personally not the hugest fan because it really makes it obvious when anything uses 'cd()' in their code without us knowing about it, and it could possibly include parse transforms and whatnot. At the same time I'm possibly acting paranoid about this and it could be no big deal.